### PR TITLE
Add Buyer Phone Number Sent By Merchant to PayPalRequest

### PR DESCRIPTION
**Pull request by agedd**
### Summary of changes

- Adds a new field `userPhoneNumber` to the `BTPayPalRequest` so that the user can also pass in his/her phone number as part of both one-time and vault flows. 
- Add a new label and textfield in the demo app to allow for entering a phone number. 
- Add sanity tests

### Checklist

- [x] Added a changelog entry

- @agedd

### CODE REVIEW

File: Sources/BraintreePayPal/BTPayPalRequest.swift [hunk 1]
```diff
@@ -96,6 +96,10 @@ import BraintreeCore
     /// :nodoc: Exposed publicly for use by PayPal Native Checkout module. This property is not covered by semantic versioning.
     @_documentation(visibility: private)
     public var paymentType: BTPayPalPaymentType
+    
+    /// Optional: A user's phone number to initiate a quicker authentication flow in the scenario where the user has a PayPal account
+    /// identified with the same phone number.
+    public var userPhoneNumber: String?
```
Comment (scannillo) [inline Sources/BraintreePayPal/BTPayPalRequest.swift:102]: Does it matter if a merchant include `-` characters in this string?

File: UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift [hunk 2]
```diff
@@ -65,4 +65,13 @@ class BTPayPalVaultRequest_Tests: XCTestCase {
         XCTAssertEqual(shippingParams["country_code"], "US")
         XCTAssertEqual(shippingParams["recipient_name"], "Recipient")
     }
+    
+    func testParametersWithConfiguration_whenUserPhoneNumberIsPresent_SetsPayerPhoneNumberInRequest() {
```
Comment (scannillo) [inline UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift:69]: Nit - you can probably lump this assertion for phone number within the JSON into the test above

**Review by scannillo — APPROVED** [review]
A few take it or leave it comments, looks good!

File: Sources/BraintreePayPal/BTPayPalRequest.swift (hunk 1 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPal/BTPayPalRequest.swift:102]: I asked this same question earlier today on the android PR, sounds like Tim is tracking down an answer: https://github.com/braintree/braintree_android/pull/1003#discussion_r1601732253

Comment (agedd) [reply Sources/BraintreePayPal/BTPayPalRequest.swift:102]: great question! yup @tdchow reached out to the gateway api team and let us know that they don't have a specific format (spaces, dashes, hyphens included within the phone number) that they recommend - seems like they have a regex which takes care of handling and removing most of these additional characters.

File: UnitTests/BraintreePayPalTests/BTPayPalCheckoutRequest_Tests.swift [hunk 3]
```diff
@@ -168,4 +168,13 @@ class BTPayPalCheckoutRequest_Tests: XCTestCase {
         
         XCTAssertNil(parameters["payer_email"])
     }
+    
+    func testParametersWithConfiguration_whenUserPhoneNumberIsPresent_SetsPayerPhoneNumberInRequest() {
```
[the remark this answers is not in the document]
Comment (agedd) [reply UnitTests/BraintreePayPalTests/BTPayPalCheckoutRequest_Tests.swift]: addressed in d40404527b3b7ffc6a5d7e535ce6491f39525eea !

File: UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift (hunk 2 above)
Comment (agedd) [reply UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift]: addressed in d40404527b3b7ffc6a5d7e535ce6491f39525eea !

**Review by KunJeongPark — APPROVED** [review]

**Comment by agedd** [comment]
marking pr as draft as API requirements are being finalized

**Comment by jaxdesmarais** [comment]
Closing PR as this work was pushed out

File: Sources/BraintreePayPal/BTPayPalRequest.swift (hunk 1 above)
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalRequest.swift:102]: It looks like Android has a separate phone object for this: https://github.com/braintree/braintree_android/pull/1186. Are we able to take the same approach on iOS so the interface is the same?

File: Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift [hunk 4]
```diff
@@ -141,6 +141,10 @@ import BraintreeCore
         if let userAuthenticationEmail, !userAuthenticationEmail.isEmpty {
             checkoutParameters["payer_email"] = userAuthenticationEmail
         }
+        
+        if let userPhoneNumber, !userPhoneNumber.isEmpty {
+            checkoutParameters["phone_number"] = userPhoneNumber
+        }
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift:147]: Can we move this logic to the `BTPayPalRequest` directly so we don't need to set it both here and in the vault request? Not sure what level it needs to be set at so may not be possible.

Comment (agedd) [reply Sources/BraintreePayPal/BTPayPalCheckoutRequest.swift]: yep, addressed in 53255318f9fa40fa7fdc6ba72359ba04ac75a495

File: Demo/Application/Features/PayPalWebCheckoutViewController.swift [hunk 5]
```diff
@@ -24,6 +24,32 @@ class PayPalWebCheckoutViewController: PaymentButtonBaseViewController {
...
+    lazy var countryCodeLabel: UILabel = {
+        let label = UILabel()
+        label.text = "Country Code:"
+        return label
+    }()
+    
+    lazy var countryCodeTextField: UITextField = {
+        let textField = UITextField()
+        textField.placeholder = "1"
+        textField.backgroundColor = .systemBackground
```
Comment (richherrera) [inline Demo/Application/Features/PayPalWebCheckoutViewController.swift:36]: what do you think about changing the keyboard type to phonePad?

`textField.keyboardType = .phonePad
`

File: Demo/Application/Features/PayPalWebCheckoutViewController.swift [hunk 6]
```diff
@@ -24,6 +24,32 @@ class PayPalWebCheckoutViewController: PaymentButtonBaseViewController {
...
+    lazy var nationalNumberLabel: UILabel = {
+        let label = UILabel()
+        label.text = "National Number:"
+        return label
+    }()
+    
+    lazy var nationalNumberTextField: UITextField = {
+        let textField = UITextField()
+        textField.placeholder = "000-000-000"
+        textField.backgroundColor = .systemBackground
```
Comment (richherrera) [inline Demo/Application/Features/PayPalWebCheckoutViewController.swift:49]: same here

`textField.keyboardType = .phonePad`

**Review by richherrera — APPROVED** [review]

File: Demo/Application/Features/PayPalWebCheckoutViewController.swift (hunk 5 above)
Comment (agedd) [reply Demo/Application/Features/PayPalWebCheckoutViewController.swift:36]: i like that! addressed in 52d24ce4639d984f6cc0f46592a763791e71753e

**Pull request merged**